components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1285,8 +1285,15 @@ where
                     self.maybe_destroy();
                 }
             }
-            CasualMessage::Campaign => {
-                let _ = self.fsm.peer.raft_group.campaign();
+            CasualMessage::Campaign { notify_by_parent } => {
+                if notify_by_parent && !self.fsm.peer.has_valid_leader() {
+                    // If the message is sent by the parent, it means that the parent is already the
+                    // leader of the parent region. And only if the split region has no leader, will
+                    // it be safe to campaign.
+                    let _ = self.fsm.peer.maybe_campaign(true);
+                } else {
+                    let _ = self.fsm.peer.raft_group.campaign();
```

Comment:
I think we should also check `self.fsm.peer.raft_group.raft.leader_id == INVALID_ID`, then it can safely start a new campaign, otherwise this campaign may lead to corrupt raft state because there may be more than 1 leader.

Reply:
Yep, it's necessary and already supplemented in the latest commit.